### DISCUSSION TITLE
Getting issue in Webpack 5.80.0 - ERROR in main Module not found: Error: Can't resolve './src' in 'project'

### DISCUSSION THREAD

**Discussion by Kranthi32** [webpack/webpack#17091] [General]
Please help me on this issue, 

ERROR in main
Module not found: Error: Can't resolve './src' in 'project'
resolve './src' in 'project'
  using description file: project\package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: project\package.json (relative path: ./src)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        project\src doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        project\src.js doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        project\src.json doesn't exist
      .wasm
        Field 'browser' doesn't contain a valid alias configuration
        project\src.wasm doesn't exist
      as directory
        project\src doesn't exist

webpack 5.80.0 compiled with 1 error in 71 ms

**Comment by alexander-akait** [comment]
You need to create ``src/index.js` file

**Comment by Kranthi32** [comment]
Hi,
My project Structure is Project/Client - this contain modules,plugins - this completely client side code . in this Client folder i created a src folder in that i added index.js file with console statement and project/server contains node code. but event adding src index.js also getting same issue. please help on this.

**Reply by alexander-akait** [reply]
Change your entries

**Comment by Kranthi32** [comment]


**Reply by alexander-akait** [reply]
Please create reproducible example using github repo, it is not readable

**Comment by Kranthi32** [comment]
please check my webpack config file .. https://github.com/Kranthi32/webpack
please suggest how to resolve this issue

**Reply by alexander-akait** [reply]
You don't need this https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L474

Do you really build an application for `node` target https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L30, please read getting started about webpack configuration

https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L141 I don't think it is a good regexp

file-loader (https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L162 ) is depreacted in favor https://webpack.js.org/guides/asset-modules/

And you really need this https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L450, I recommend to keep defualt webpack value (i.e. don't provide value)

Don't use only one name here https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L26, use `[name].js`

**Reply by alexander-akait** [reply]
You don't need it https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L32, if your target is web

**Comment by Kranthi32** [comment]
As per your suggestion iam updating webpack configration, but server code not generating and getting another Error below.

[webpack-cli] [Error: ENOTEMPTY: directory not empty, rmdir 'Project\dist\Client\Plugins\Style\images'] {
  errno: -4051,
  code: 'ENOTEMPTY',
  syscall: 'rmdir',
  path: 'Project\\dist\\Client\\Plugins\\Style\\images'
}

and this is required why because my server code is node js 

Do you really build an application for node target https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L31, please read getting started about webpack configuration

**Reply by alexander-akait** [reply]
Sorry, I am not magic and can't help only have an error message, you have a stack trace so you can see there is a problem place

**Comment by Kranthi32** [comment]
Hi, Here i am adding my sample project. this project contain login and Home pages. dist file creating while running the running the application through is working fine but getting below error. Please help me on this.

Project: https://github.com/Kranthi32/webpack

Error:  ERROR in main
Module not found: Error: Can't resolve './src' in 'C:\Users\102583\Desktop\myproject'
resolve './src' in 'C:\Users\102583\Desktop\myproject'
  using description file: C:\Users\102583\Desktop\myproject\package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: C:\Users\102583\Desktop\myproject\package.json (relative path: ./src)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\102583\Desktop\myproject\src doesn't exist
      .js
        C:\Users\102583\Desktop\myproject\src.js doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\102583\Desktop\myproject\src.json doesn't exist
      .wasm
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\102583\Desktop\myproject\src.wasm doesn't exist
      as directory
        C:\Users\102583\Desktop\myproject\src doesn't exist

webpack 5.81.0 compiled with 1 error in 135 ms

**Comment by Kranthi32** [comment]
And one more issue css files are not creating if i comment copywebpackplugin of css ,  at https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L200

**Reply by alexander-akait** [reply]
> module.exports = [clientConfig, serverConfig, customConfig];

but, `customConfig` contains only:
```
const customConfig = {
    stats: {
        colors: true,
        modules: true,
        reasons: true,
        errorDetails: true
    },
}
```

Why do you have the compilation only with the `stats` property? Remove it, if you want to use this configuration in `client` and `server` config, just use `...customConfig `

**Reply by Kranthi32** [reply]
Custom config i created for findout  webpack errors. After removing its working. Thanks for the suggestionMy question is why css files are not creating from this line in webpack config
https://github.com/Kranthi32/webpack/blob/main/webpack.config.js#L200 need to take input css files from client/styles to dist/styles folder or else how to provide separate separate css file names in config.

**Reply by alexander-akait** [reply]
What do you mean don't work?

**Reply by Kranthi32** [reply]
how to provide multiple entry points for css files with proper input path and how to save those css files speciafical output folder

**Reply by alexander-akait** [reply]
Sorry, I don't understand you, there are a lot of documentation, please read it carefully, if you need bundle CSS files (i.e. resolve `url(...)` and `@import`) you need to use `css-loader` + `mini-css-extract-plugin`/`style-loader`, if you want just copy files to dist you can use `copy-webpack-plugin`

**Comment by Judahmeek** [comment]
To anyone here from a search engine, `ERROR in main Module not found: Error: Can't resolve './src'` likely means that one or more of your configured `entry` values results in `undefined` & webpack is trying to use [its default entry value](https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L174), which doesn't exist in your codebase.
